process/transaction/process.go

```diff
@@ -75,23 +92,101 @@ func (txProc *txProcessor) ProcessTransaction(tx *transaction.Transaction, round
 		return err
 	}
 
-	txType, err := txProc.scProcessor.ComputeTransactionType(tx)
+	txType, err := txProc.txTypeHandler.ComputeTransactionType(tx)
 	if err != nil {
 		return err
 	}
 
 	switch txType {
 	case process.MoveBalance:
-		return txProc.processMoveBalance(tx, adrSrc, adrDst)
+		currTx, ok := tx.(*transaction.Transaction)
```

Comment:
Each function could make it's own type assertion, then this switch would be less verbose

Reply:
done